pkg/api/dashboard.go

```diff
@@ -515,12 +556,22 @@ func (hs *HTTPServer) postDashboard(c *contextmodel.ReqContext, cmd dashboards.S
 // 401: unauthorisedError
 // 500: internalServerError
 func (hs *HTTPServer) GetHomeDashboard(c *contextmodel.ReqContext) response.Response {
-	prefsQuery := pref.GetPreferenceWithDefaultsQuery{OrgID: c.OrgID, UserID: c.SignedInUser.UserID, Teams: c.Teams}
+	namespaceID, userIDstr := c.SignedInUser.GetNamespacedID()
+
+	if namespaceID != identity.NamespaceUser {
+		return response.Error(http.StatusBadRequest, "User does not belong to a user namespace", nil)
+	}
+
```

Comment:
this is a strange check, why is this check here in GetHomeDashboard? it looks like a very generic middleware / auth check? 

Oh and it breaks anonymous access

Reply:
https://github.com/grafana/grafana/pull/74397 this should cover a lot of the issues with anonymous access but I'll go through this one restoring ID 0 being passed to stores until we can have better checks on the store